### PR TITLE
Add Rollback template to script 

### PR DESCRIPTION
**Pull request by stevieraykatz**
TLDR; this PR implements an example script which shows that a rollback transaction should explicitly set the nonce returned by `_getNonce()` using an .env var. Hopefully this template will help avoid issues seen below. 

Full context: 

In executing `mainnet/2024-03-05-pause-unpause-test` we ran into an issue wherein the signatures were being incorrectly ordered by the execution call. This was leading to reverts since the Safe requires that signatures are arranged in ascending order.

 The root cause was traced back to the fact that `_getNonce` was being overwritten by the following: 
```solidity
    function _getNonce(IGnosisSafe safe) internal override view returns (uint256 nonce) {
        uint256 _nonce = safe.nonce();
        console.log("Safe current nonce:", _nonce);
        console.log("Incrementing by 1 to account for planned `Pause` tx");
        return _nonce+1;
    }
```
This was fine for signing and simulating the transaction because these occurred before the preceding `Pause` transaction had been submitted. But when it came time to execute the transaction, this nonce increment was used in the execution context causing the `hash` used by `ecrecover` to differ from the correct `hash`. In turn, this lead to invalid addresses being returned which were then used to sort the signatures incorrectly.

### CODE REVIEW

**Review by OKEAMAH — APPROVED** [review]
Cool

**Comment by cb-heimdall** [comment]
Review Error for OKEAMAH @ 2024-07-13 21:31:35 UTC
User must have write permissions to review
